XamarinCommunityToolkit/Views/StateLayout/State.shared.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Xamarin.CommunityToolkit.UI.Views
+{
+	public enum State
+	{
+		None,
+		Loading,
+		Saving,
+		Success,
+		Error,
+		Empty,
+		Custom
```

Comment:
I wonder, is there any sense to keep enum if we cannot describe all possible cases?

Maybe we can refactor it to static class (similar to how XF Device class operates with RuntimePlatform)
```csharp
public static class LayoutState {
    public const string None = nameof(None);
    public const string Loading = nameof(Loading);
    /// etc.
}
```

Reply:
Been trying this but so far it's been a bunch of misery :P The custom state is explicitly added for scenarios not covered by the default states. Also making it a string introduces some strange scenarios where the user **has to** explicitly set a value for it to work. We can fix that in the component obviously, but it feels a bit strange.